CONTRIBUTING.md

```diff
@@ -1,20 +1,144 @@
+
+If you'd like to work on a bug, please comment on it to let the maintainers know.
+If someone else has already commented and taken up that bug, please refrain from working on it and submitting
+a PR without asking the maintainers as it leads to unnecessary duplication of effort.
+
+
+# Installation
+
+To get started on a patch, first install `addons-linter` from [source](README.md#development).
+
+## Develop all the things
+
+Your one stop command to continuously run tests and check for
+JavaScript syntax problems.
+
+    npm test
+
+The other commands below are just variations on this.
+
+## Run all application tests
+
+To run the entire suite of tests once and exit, type:
+
+    npm run test-once
+
+This is the same as the `npm test` command but it won't re-run automatically as
+you edit files.
+
+### Run a single test
+
+Instead of running the entire suite, you can run a single test by invoking
+the `jest` executable directly with the `-t` option to filter by test
+description. For example, if the test you'd like to run is defined in
+`tests/test.linter.js` and is described as
+"should collect an error when not an xpi/zip" then you could run it like this:
+
+    ./node_modules/.bin/jest -r tests/test.linter.js -f "not an xpi/zip"
+
+### Debug a test
+
+You can enter the [Node debugger](https://nodejs.org/api/debugger.html) by
+directly invoking the `npm run debug` command. For example,
+if the test you want to debug is defined in `tests/test.linter.js` then you
+could enter the debugger like this:
+
+    node --inspect --inspect-brk ./node_modules/.bin/jest tests/test.linter.js -t 'flag potentially minified'
+
+You could also put the `debugger` statement somewhere in the code to set a
+breakpoint.
+
+## Build addons-linter
+
+Type `npm run build` to build a new version of the libraries used by the
+`./bin/addons-linter` command. When successful, you will see newly built files in
+the `./dist/` directory.
+
+## Creating a pull request
+
+When you create a
+[pull request](https://help.github.com/articles/creating-a-pull-request/)
+for a new fix or feature, be sure to mention the issue
+number for what you're working on.
+The best way to do it is to mention the issue like
+this at the top of your description:
+
+    Fixes #123
+
+The issue number in this case is "123."
+The word *Fixes* is magical; github will automatically close the issue when your
+pull request is merged.
+
+## Writing commit messages
+
+Good commit messages serve at least three important purposes:
+
+* To speed up the reviewing process.
+* To help us write a good release note.
+* To help the future maintainers of addons-linter (it could be you!), say five years into the future, to find out why a particular change was made to the code or why a specific feature was added.
```

Comment:
This sentence is too long especially compared to the others. I'd just say "They help future maintainers understand your change and the reasons behind it."